File: Nov2020/29Nov/Star-studdedLockdown.cpp
/*
Problem Name: Star-studded Lockdown
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/star-studded-lockdown/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int t;
    cin>>t;
    while(t--){
        int n;
        long sum=0;
        cin>>n;
        string s;
        map<char,int> m;
        cin>>s;
        for(int i=0;i<s.length();++i){
            sum+=m[s[i]];
            m[s[i]]++;
        }
        cout<<sum<<"\n";
    }
    return 0;
}

File: Nov2020/29Nov/Star-StuddedLockdown.java
/*
Problem Name: Star-studded Lockdown
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/star-studded-lockdown/
*/
import java.util.*;
import java.io.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int t = in.nextInt();
        for(;t>0;t--){
            int n = in.nextInt();
            char arr[] = in.next().toCharArray();
            int cnt[] = new int[26];
            long sum = 0;
            for(int i=0;i<n;i++){
                sum += cnt[arr[i] - 'a'];
                cnt[arr[i] - 'a']++;
            }
            out.println(sum);
        }
        out.flush();
    }
}
